src/Workspaces/Remote/ServiceHub/Services/AssetSynchronization/RemoteAssetSynchronizationService.cs

```diff
@@ -49,24 +49,39 @@ public ValueTask SynchronizeActiveDocumentAsync(DocumentId? documentId, Cancella
     }
 
     public ValueTask SynchronizeTextChangesAsync(
-        ImmutableArray<(DocumentId documentId, Checksum baseTextChecksum, ImmutableArray<TextChange> textChanges, Checksum newTextChecksum)> changes,
+        DocumentId documentId,
+        Checksum baseTextChecksum,
+        ImmutableArray<TextChange> textChanges,
+        Checksum newTextChecksum,
         CancellationToken cancellationToken)
     {
         return RunServiceAsync(async cancellationToken =>
         {
-            var workspace = GetWorkspace();
+            var wasSynchronized = await SynchronizeTextChangesHelperAsync().ConfigureAwait(false);
 
-            using (RoslynLogger.LogBlock(FunctionId.RemoteHostService_SynchronizeTextAsync, cancellationToken))
+            var metricName = wasSynchronized ? "SucceededCount" : "FailedCount";
+            TelemetryLogging.LogAggregatedCounter(FunctionId.RemoteHostService_SynchronizeTextAsyncStatus, KeyValueLogMessage.Create(m =>
             {
-                foreach (var (documentId, baseTextChecksum, textChanges, newTextChecksum) in changes)
+                m[TelemetryLogging.KeyName] = nameof(RemoteAssetSynchronizationService) + "." + metricName;
```

Comment:
Consider constants for the two values so we don't keep allocating dupe strings.

Reply:
Done!